### ISSUE TITLE
Inputs doesn't allow attributes now

### ISSUE THREAD

**Issue by juliogarciag** [activeadmin/activeadmin#3954]
I updated activeadmin from an old github commit from the past year to the latest version and i found that the `inputs` method in the form doesn't receive html attributes anymore. Previously, this was working:

```ruby
f.inputs('Password Fields', 'ng-if' => 'changePassword') do
  f.input :password, :as => :password
  f.input :password_confirmation, :as => :password
end
```

But now, i can't find a way to pass the 'ng-if' attribute (required by angular) to the fieldset. There is another way to do that without rewriting the markup for the `inputs` everytime? (i have dozens of usages of attributes as the `inputs` second parameter)

I think this happens because of [this method](https://github.com/activeadmin/activeadmin/blob/0eee8162142d11a1296fdbfb5118a32d6beb09e1/lib/active_admin/views/components/active_admin_form.rb#L97) hardcoding the fieldset construction.

**Comment by blackjid** [comment]
i'm having the same problem.....

**Comment by bunzli** [comment]
+1

**Comment by osyoav2** [comment]
Hi All.
I've looked into this and it seems more like a formtastic issue rather then AA one.
It seems "inputs" method supports only standard attributes according to the docs [here] (https://github.com/justinfrench/formtastic/wiki/4.1-Options-Cheat-Sheet).
@timoschilling tried to refactor the inputs method in issue https://github.com/justinfrench/formtastic/issues/1089 but I don't think it came through.
Maybe @justinfrench can help?

**Comment by justinfrench** [comment]
Hello, here's a few lines from the rdoc for `Formtastic::Helpers::InputsHelper`:

```
# All options except `:name`, `:title` and `:for` will be passed down to the fieldset as HTML
# attributes (id, class, style, etc).
```

And here's some relevant lines from `inputs_spec_helper.rb`:

```ruby
    describe 'when other options are provided' do
      before do
        @id_option = 'advanced'
        @class_option = 'wide'
  
        concat(semantic_form_for(@new_post) do |builder|
          builder.inputs :id => @id_option, :class => @class_option do
          end
        end)
      end
  
      it 'should pass the options into the fieldset tag as attributes' do
        output_buffer.should have_tag("form fieldset##{@id_option}")
        output_buffer.should have_tag("form fieldset.#{@class_option}")
      end
    end
```

The key line is `:class => @class_option`, so I see no reason why this shouldn't work in a Formtastic-only scenario. It could be a bug in Formtastic, but having ActiveAdmin layered in creates a lot of open questions, as I don't use it. I'd encourage you to:

1. Work-up an isolated, Formtastic-only test-case
2. Report an issue on Formtastic directly if there's a bug, or 
3. Work-up an isolated, ActiveAdmin test case if you've ruled out Formtastic

**Comment by nileshtrivedi** [comment]
I too am facing this. Trying to use "disabled" attribute on a fieldset to disable a whole bunch of inputs at once upon a certain condition. But inputs method doesn't seem to support input_html, or html, or any other parameter which can set HTML attributes.

**Comment by onyxraven** [comment]
the OP has the right code link.  It looks like AA is bypassing Formtastic's opening tag and inserting its own, only pulling in the 'class' attribute.  Whereas Formtastic does [quite a bit of work](https://github.com/justinfrench/formtastic/blob/master/lib/formtastic/helpers/inputs_helper.rb) with extra attributes.

**Comment by afurmanov** [comment]
The AA does not always behaves like that, when `inputs` has block with arity > 0 it falls back to Formtastic behavior. I described workaround here: https://github.com/activeadmin/activeadmin/issues/4516

**Comment by timoschilling** [comment]
fixed by #4518
